design/Extend-VolumePolicies-to-support-more-actions.md

```diff
@@ -0,0 +1,161 @@
+# Extend VolumePolicies to support more actions
+
+## Abstract
+
+Currently, the [VolumePolicies feature](https://github.com/vmware-tanzu/velero/blob/main/design/Implemented/handle-backup-of-volumes-by-resources-filters.md) which can be used to filter/handle volumes during backup only supports the skip action on matching conditions. Users need more actions to be supported.
+
+## Background
+
+The `VolumePolicies` feature was introduced in Velero 1.11 as a flexible way to handle volumes. The main agenda of introducing the VolumePolicies feature was to improve the overall user experience when performing backup operations for volume resources, the feature enables users to group volumes according the `conditions` (criteria) specified and also lets you specify the `action` that velero needs to take for these grouped volumes during the backup operation. The limitation being that currently `VolumePolicies` only supports `skip` as an action, We want to extend the `action` functionality to support more usable options like `fs-backup-volumes` (File system backup) and `fs-backup-volumes-excludes` (VolumeSnapshots).
+
+## Goals
+- Extending the VolumePolicies to support more actions like file system backup(`fs-backup-volumes`) and snapshots(`fs-backup-volumes-excludes`).
```

Comment:
@shubham-pampattiwar 
`fs-backup-volumes-excludes` may not be suitable for a snapshot way of backup, This name may have categories that are too general and the user may not related to `volume snapshot`; we need to make them more specific.

and also if we extend another more way of backup, this name would be more suitable, maybe a specific one is better

Reply:
Basically a volume matching this policy behaves exactly the same as if that volume had the backup-volumes-excludes annotation. It is opted out of fs-backup when we default to fs backup. That is exactly the scope of this new policy, so a more specific name would be less accurate.

Reply:
yes agreed with @sseago. @qiuming-best let me know what name feels accurate to you ? We are not gonna say `volume-snapshot` as an action for sure. Does the dropping for `fs-` prefix work for you ?